pkg/api/vlabs/types.go

```diff
@@ -195,10 +195,12 @@ type AgentPoolProfile struct {
 	DiskSizesGB         []int  `json:"diskSizesGB,omitempty"`
 	VnetSubnetID        string `json:"vnetSubnetID,omitempty"`
 	IPAddressCount      int    `json:"ipAddressCount,omitempty"`
-
+	OsImageOffer      	string `json:"osImageOffer,omitempty"`
```

Comment:
- The flattened approach to providing offer, sku, version (versus being in a struct) in agentpool is concerning. Also, not having this in LinuxProfile doesn't seem correct (and has long term impact to the API model).
- I see that masters are left to be Ubuntu which might be ok (and convenient) in the short terms but introduces complexity around interactions between agents and masters & what happens when master will be CoreOS as well.
- Also, what about testing? How do we make sure this feature doesn't fall behind in the test matrix?